prowler/providers/common/test_connection_dataclass.py

```diff
@@ -0,0 +1,9 @@
+from dataclasses import dataclass
+from typing import Any
+
+
+@dataclass
+class TestConnection:
+    connected: bool = False
+    error: Exception = None
+    result: Any = None
```

Comment:
I'm curious how the `result` should be used? 

Is it needed? For all providers?

Reply:
Not needed at the end, since this method is just a `test-connection makes no sense to return credentials, it is not a `setup_credentials` method.